CMakeLists.txt

```diff
@@ -9,30 +9,49 @@ project(quicr-transport
 if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
     option(QTRANSPORT_BUILD_TESTS "Build tests for quicr transport" ON)
 else()
-    option(QTRANSPORT_BUILD_TESTS "Build tests for quicr trasnsport" OFF)
+    option(QTRANSPORT_BUILD_TESTS "Build tests for quicr transport" OFF)
 endif()
 
+option(QUIC_TRANSPORT "Enable QUIC Transport (Default ON)" ON)
+option(PLATFORM_ESP_IDF "Enabble suppport for esp-idf (Default OFF" OFF)
 
 find_package(Threads REQUIRED)
 find_package(PkgConfig REQUIRED)
 
 ###
 ### Dependencies
 ###
-add_subdirectory(dependencies)
+if (QUIC_TRANSPORT)
+  add_subdirectory(dependencies)
+endif()
 
 ###
 ### Main code
 ###
-add_library(quicr-transport
-        src/transport.cpp
-        src/transport_udp.cpp
-        src/transport_picoquic.cpp
-        )
-
-target_link_libraries(quicr-transport
-        PUBLIC
-        picoquic-core)
+
+set (sources 
+    src/transport.cpp
+    src/transport_udp.cpp
+)
+
+if (QUIC_TRANSPORT)
+    list(APPEND sources src/transport_picoquic.cpp)
+endif()
+
+
+add_library(quicr-transport ${sources})
+
+if (QUIC_TRANSPORT)
+  target_link_libraries(quicr-transport
+          PUBLIC
+          picoquic-core)
+   add_compile_definitions(${LIB_NAME} ENABLE_QUIC)       
+endif()          
+
+if (PLATFORM_ESP_IDF)
```

Comment:
```/Users/timevens/Documents/workspace/ws-media10x/transport/src/transport_udp.cpp:20:10: fatal error: 'lwip/netdb.h' file not found```

Looks like we need to include submodule with the new headers used.

Reply:
this option is enabled when building with esp framework and not as submodules. So we never enable this option for anything outside that setup and hence would be ok.

Reply:
What's the make file or other syntax to build this?  Right now it doesn't build for me.

Reply:
you can't build it unless you are building within esp framework. As long as transport builds with no PLATFORM_ESP_IDF,  we should be fine to merge this PR